week3/RESTfulAPIs/app.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+const express = require('express');
+const bodyParser = require('body-parser');
+const fs = require('fs');
+const uniqid = require('uniqid');
+
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(bodyParser.urlencoded({
+    extended: false
+}));
+app.use(bodyParser.json());
+
+const TODOS_FILENAME = './data/todo.json';
+let myTodoList = [];
+//console.log('my to do list  : ',myTodoList);
+app.listen(port, (err) => {
```

Comment:
You run your 'app.listen()` here, but you don't create your routes until line 102 which is inside a Promise.then().

Here's a potential problem:
Imagine that the todos file becomes 100GB in size. Now, it takes quite a long time for your `readTodosFile` function to run.
Since you start listening here, but set up the routes AFTER you've read the todo file, there might be a long time where your server is listening but the routes haven't been created (since the todo file takes a long time to read).

You should either put the call to `readTodosFile` somewhere else, like inside the routes (it's important to think about WHEN functions get executed and lines of code get hit), or put your `app.listen` inside that same Promise.then, so that the server doesn't start listening until it's actually ready.